### PR TITLE
Document gevent concurrency

### PR DESCRIPTION
**Pull request by cunla**
## Description
Add documentation for gevent, following [this issue](https://github.com/celery/celery/issues/8425#issuecomment-1701058121)

### CODE REVIEW

File: docs/userguide/concurrency/gevent.rst [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+simpler implementation and better performance. Read why others `use
+gevent`_ and check out the list of the `open source projects based on
+gevent`_.
+
+
+Enabling gevent
+=================
+
+
+You can enable the Eventlet pool by using the
```
Comment (auvipy) [inline docs/userguide/concurrency/gevent.rst:42]:
```suggestion
You can enable the gevent pool by using the
```

Comment (auvipy) [reply docs/userguide/concurrency/gevent.rst:42]: Eventlet pool or gevent pool?

**Review by auvipy — CHANGES_REQUESTED** [review]

**Review by Nusnus — APPROVED** [review]

**Comment by Nusnus** [comment]
Great addition to the docs!
Well done 💯 - liked the links to the known issues, useful.

Thank you! @cunla

File: docs/userguide/concurrency/gevent.rst [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+gevent is `inspired by eventlet`_ but features a more consistent API,
+simpler implementation and better performance. Read why others `use
+gevent`_ and check out the list of the `open source projects based on
+gevent`_.
+
+
+Enabling gevent
+=================
+
```
Comment (auvipy) [inline docs/userguide/concurrency/gevent.rst:40]:
```suggestion
```

File: docs/userguide/concurrency/gevent.rst [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+Introduction
+============
+
+The `gevent`_ homepage describes it a coroutine_ -based Python_ networking library that uses
+`greenlet <https://greenlet.readthedocs.io>`_ to provide a high-level synchronous API on top of the `libev`_
+or `libuv`_ event loop.
+
+Features include:
+
```
Comment (auvipy) [inline docs/userguide/concurrency/gevent.rst:17]:
```suggestion
```

File: docs/userguide/concurrency/gevent.rst [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+See the `gevent examples`_ directory in the Celery distribution for
+some examples taking use of Eventlet support.
+
+Known issues
+============
+There is a known issue using python 3.11 and gevent.
+The issue is documented `here`_ and addressed in a `gevent issue`_.
+Upgrading to greenlet 3.0 solves it.
+
```
Comment (auvipy) [inline docs/userguide/concurrency/gevent.rst:63]:
```suggestion
```

File: examples/gevent/README.rst [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    $ python
+    >>> from tasks import urlopen
+    >>> from celery import group
+    >>> result = group(urlopen.s(url)
+    ...                     for url in LIST_OF_URLS).apply_async()
+    >>> for incoming_result in result.iter_native():
+    ...     print(incoming_result)
+
+
+
```
Comment (auvipy) [inline examples/gevent/README.rst:49]:
```suggestion
```

**Review by auvipy — CHANGES_REQUESTED** [review]

**Review by auvipy — APPROVED** [review]

**Pull request merged**